### PR TITLE
Added autoscaling docs

### PR DESCRIPTION
**Pull request by EmanekaT**
* Add documentations for new HPA configurations. docs/source/hpa.rst and docs/source/autoscaling.rst
* Fixed a couple HPA bugs.  paasta_tools/kubernetes/application/controller_wrappers.py
* Upgraded Sphinx to 2.20 and changed to read-the-docs theme, which looks much better. That's why there are so many changed generated files. Feel free to try it yourself.

### CODE REVIEW

File: docs/source/hpa.rst [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  **Please make sure your dimensions are different from the dimensions used others**
+  HPAMetrics Collector will collect metrics from your HTTP endpoints, add your specified
+  dimensions, and send them to SignalFX. So please make sure your service won't cause troubles
+  for other people.
+
+uwsgi
+^^^^^
+ Same as http, but "/status/uwsgi" is used as the endpoint.
+
+your-signalfx-own-metrics
```
Comment (EvanKrall) [inline docs/source/hpa.rst]: Make sure this matches the actual YAML provided (`your-own-sfx-metrics` or `another-random-sfx-metrics`).

File: docs/source/hpa.rst [hunk 2]
```diff
@@ -0,0 +1,159 @@
+.. _hpa:
+====================================
+Autoscaling Kubernetes pods in PaaSTA
+====================================
+
+This section introduces a new configuration for services running on Kubernetes.
+It allows developers to utilize the full power of the new autoscaling system on
+Kubernetes. If you are interested in the architectural design, please read `this blog post <https://yelpwiki.yelpcorp.com/display/PAASTA/2019/12/09/Service+Autoscaling+with+HPA>`_
```
[the remark this answers is not in the document]
Comment (EmanekaT) [reply docs/source/hpa.rst]: image
I think it shouldn't matter anymore in read-the-docs theme.

File: docs/source/hpa.rst [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+
+Alrogithms
```
[the remark this answers is not in the document]
Comment (EmanekaT) [reply docs/source/hpa.rst]: Nice catch. I should check it in Google docs first.

File: docs/source/hpa.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (EvanKrall) [reply docs/source/hpa.rst]: Correct, line breaks shouldn't matter once rendered. One-sentence-per-line is for the benefit of people editing the .rst source.

File: docs/source/generated/paasta_tools.api.views.deploy_queue.rst [hunk 4]
```diff
@@ -1,7 +1,17 @@
+<<<<<<< HEAD
```
Comment (EvanKrall) [inline docs/source/generated/paasta_tools.api.views.deploy_queue.rst]: looks like some merge conflict markers showedu p

File: docs/source/generated/paasta_tools.spark_tools.rst [hunk 5]
```diff
@@ -1,7 +1,17 @@
...
     :show-inheritance:
+=======
+paasta\_tools.spark\_tools module
+=================================
+
+.. automodule:: paasta_tools.spark_tools
+   :members:
+   :undoc-members:
+   :show-inheritance:
+>>>>>>> added leftovers
```
Comment (EvanKrall) [inline docs/source/generated/paasta_tools.spark_tools.rst]: more merge conflict markers

**Pull request merged**